test/e2e/network_policy.go

```diff
@@ -0,0 +1,509 @@
+			ObjectMeta: api.ObjectMeta{
+				Name: "test-policy",
+			},
+			Spec: extensions.NetworkPolicySpec{
+				// Apply to server
+				PodSelector: unversioned.LabelSelector{
+					MatchLabels: map[string]string{
+						"pod-name": podServer.Name,
+					},
+				},
+				// Allow traffic only to one port.
+				Ingress: []extensions.NetworkPolicyIngressRule{{
+					Ports: []extensions.NetworkPolicyPort{{
+						Port: &intstr.IntOrString{IntVal: listeningPort1},
+					}},
+				}},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			By("Cleaning up the policy")
+			if err := f.Client.NetworkPolicies(ns.Name).Delete(policy.Name, nil); err != nil {
+				framework.Failf("unable to delete policy %v: %v", policy.Name, err)
+			}
+		}()
+
+		By("Creating client-a that should succesfully connect to the opened port.")
+		podClientA := createClientPod(f, ns, clientAName, service, listeningPort1)
+		defer func() {
+			By("Cleaning up the pod Server A")
+			if err := f.Client.Pods(ns.Name).Delete(podClientA.Name, nil); err != nil {
+				framework.Failf("unable to delete pod %v: %v", podClientA.Name, err)
+			}
+		}()
+
+		By("Creating a client-b that should not succesfully connect to the closed port.")
+		podClientB := createClientPod(f, ns, clientBName, service, listeningPort2)
+		defer func() {
+			By("Cleaning up client-b.")
+			if err := f.Client.Pods(ns.Name).Delete(podClientB.Name, nil); err != nil {
+				framework.Failf("unable to delete pod %v: %v", podClientB.Name, err)
+			}
+		}()
+
+		framework.Logf("Waiting for both clients to run.")
+		err = framework.WaitForPodRunningInNamespace(f.Client, podClientA)
+		Expect(err).NotTo(HaveOccurred(), "waiting for Client-A to run")
+		err = framework.WaitForPodRunningInNamespace(f.Client, podClientB)
+		Expect(err).NotTo(HaveOccurred(), "waiting for Client-B to run")
+
+		framework.Logf("Waiting for client-a to complete.")
+		err = framework.WaitForPodSuccessInNamespace(f.Client, podClientA.Name, ns.Name)
+		Expect(err).NotTo(HaveOccurred(), "checking Client-A could communicate with server.")
+
+		framework.Logf("Waiting for client-b to complete.")
+		err = framework.WaitForPodSuccessInNamespace(f.Client, podClientB.Name, ns.Name)
+		Expect(err).To(HaveOccurred(), "checking Client-B could not communicate with server")
+	})
+
+	It("should enforce policy based on NamespaceSelector [Feature:NetworkPolicy]", func() {
+		nsA := f.Namespace
+		nsBName := f.BaseName + "-b"
+		// The CreateNamespace helper uses the input name as a Name Generator, so the namespace itself
+		// will have a different name than what we are setting as the value of ns-name.
+		// This is fine as long as we don't try to match the label as nsB.Name in our policy.
+		nsB, err := f.CreateNamespace(nsBName, map[string]string{
+			"ns-name": nsBName,
+		})
+		Expect(err).NotTo(HaveOccurred())
+		setNamespaceIsolation(f, nsA, "DefaultDeny")
+		setNamespaceIsolation(f, nsB, "DefaultDeny")
```

Comment:
Strictly speaking, this test doesn't need isolation on Namespace B (since it's just testing policy in Namespace A).